wakuv2/waku.go

```diff
@@ -1831,14 +1838,22 @@ func FormatPeerStats(wakuNode *node.WakuNode, peers node.PeerStats) map[string]t
 	return p
 }
 
-func formatConnStatus(wakuNode *node.WakuNode, c node.ConnStatus) types.ConnStatus {
+func formatConnStatus(wakuNode *node.WakuNode, c peermanager.TopicHealthStatus) types.ConnStatus {
+	isOnline := true
+	if c.Health == peermanager.UnHealthy {
+		isOnline = false
+	}
+
 	return types.ConnStatus{
-		IsOnline:   c.IsOnline,
-		HasHistory: c.HasHistory,
-		Peers:      FormatPeerStats(wakuNode, c.Peers),
+		IsOnline: isOnline,
+		Peers:    FormatPeerStats(wakuNode),
 	}
 }
 
-func (w *Waku) StoreNode() store.Store {
+func (w *Waku) StoreNode() *store.WakuStore {
 	return w.node.Store()
 }
+
+func (w *Waku) LegacyStoreNode() legacy_store.Store {
```

Comment:
Any specific reason for still having legacyStoreNode?

Reply:
One of the test units require running a store node.
I imagine it could be possible to just run docker for that, and get rid completely of the legacy store node. I just don't know if it's worth the effort since it's only for a single test unit.